### PR TITLE
download music option when on Music View

### PR DESCRIPTION
**Pull request by MiTo0o**
- I kinda had to create a couple more states and pass it around, not sure if that's ok with everyone
- Downloading isn't that straightforward, I had to do a work around since downloading cross origin isn't allowed
- currently, the song download name doesn't account for songs that have a second tune, but quite a bit of work needs to be done to see if its the second tune or not
- If this looks ok, I will start writing tests

addresses #123

### CODE REVIEW

File: src/components/NavigationBar.tsx [hunk 1]
```diff
@@ -27,6 +31,7 @@ const NavigationBar: React.FC<NavigationBarProps> = (props) => {
   const [showSettingsModal, setShowSettingsModal] = useState<boolean>(false);
   const [songbookName, setSongbookName] = useState<string>(defaultNavigationTitle);
   const { bookId } = useParams<{ bookId: string }>();
+  const [songPageBlobUrl, setSongPageBlobUrl] = useState<string>("");
```
Comment (brotatotes) [inline src/components/NavigationBar.tsx:34]: remove if possible

Comment (brotatotes) [reply src/components/NavigationBar.tsx:34]: the empty string `""`

File: src/pages/SongPage.tsx [hunk 2]
```diff
@@ -23,6 +23,8 @@ import { createSwipeGesture } from "../utils/UiUtils";
 const SongPage: React.FC = () => {
   const { bookId, songId } = useParams<{ bookId: string; songId: string }>();
   const history = useHistory();
+  // music page url
+  const [musicPageUrl, setMusicPageUrl] = useState<string>("");
```
Comment (brotatotes) [inline src/pages/SongPage.tsx:27]: remove empty string `""` if possible

**Comment by brotatotes** [comment]
- let's check the behavior on mobile (app + mobile web). OR you can disable it for those for now and just get this in
- don't think we need to have separate dl file names for songs w diff tunes, this is great

**Review by brotatotes — CHANGES_REQUESTED** [review]

File: src/components/NavigationBar.tsx (hunk 1 above)
Comment (MiTo0o) [reply src/components/NavigationBar.tsx:34]: it will default to undefined, don't think we want that

File: src/pages/SongPage.tsx (hunk 2 above)
Comment (MiTo0o) [reply src/pages/SongPage.tsx:27]: will default to undefined

**Review by brotatotes — APPROVED** [review]

**Pull request merged**